arch/x86/kvm/pkvm/mmu.c

```diff
@@ -342,3 +360,80 @@ int pkvm_host_mmu_unmap(unsigned long vaddr, unsigned long size)
 	/* The vaddr == phys for the host MMU */
 	return pkvm_pgtable_unmap(&host_mmu, vaddr, vaddr, size);
 }
+
+/**
+ * pkvm_host_donate_hyp() - Donate memory pages from host to hypervisor.
+ * @phys:	Physical address of the memory region to donate.
+ * @size:	Size of the memory region to donate.
+ *
+ * This is a memory protection API under the host mmu lock protection. It
+ * donates the memory pages in range [@phys, @phys + @size) from host to hyp via
+ * pkvm_pgtable_set_owner().
+ *
+ * The donated memory pages are annotated with PKVM_ID_HYP owner id in the host
+ * mmu, to indicate the memory pages are owned by the hypervisor.
+ *
+ * Before donate, check the host memory page state to make sure the donated
+ * memory pages are PKVM_PAGE_OWNED by the host. After donate, change the host
+ * memory page state to PKVM_PAGE_NONE to indicate no mappings in host mmu.
```

Comment:
IMHO instead of mentioning all these implementation details that don't need to be mentioned (e.g. that it calls pkvm_pgtable_set_owner()), the documentation should just give a clear high-level explanation of what "donate" even means (i.e. that the pages that were owned by the host become owned by the hypervisor, and become protected from the host).

This applies to other functions (pkvm_host_donate_hyp, pkvm_host_share_hyp, pkvm_host_unshare_hyp) as well.